src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -38,28 +41,45 @@ public Reason getReason() {
         return Optional.ofNullable(cause);
     }
 
+    /**
+     * @return whether sqlQuery is recognized as writeOperation
+     */
+    public Optional<Boolean> isSqlWrite() {
```

Comment:
`Optional<Boolean>` 🤨

Reply:
It can even be an `Optional` of `null` 😬

Reply:
Shame on me!